### PR TITLE
Added Player Battles

### PR DESCRIPTION
**Pull request by BlackHoleFox**
This adds player battles to the wrapper, complying with the new API structure. Minor package.json version bump. It has been linted and it returns data as expected.

The array is wrapped in an object, I couldn't find a way to just return a raw array. Pretty sick atm so can't problem solve that well. If you have a way to, I will gladly go in and modify it to return a raw array instead of needing to access an object's property.

Cheers, Fox.

### CODE REVIEW

**Comment by BlackHoleFox** [comment]
Clan war endpoints added as well. Linted and tested to ensure functionality and ease of use. Docs have been updated accordingly as well.

File: src/structures/player.js [hunk 1]
```diff
@@ -199,6 +199,14 @@ class Player {
             };
         }
 
+        if (data.playerBattles) {
```
Comment (bsantosh909) [inline src/structures/player.js:202]: Since its already inside player class better simply name it `battles`

File: src/client.js [hunk 2]
```diff
@@ -122,6 +122,25 @@ class Client {
...
+     * })
+     *  .then(player => {
+     *    console.log(`The Player's past 25 battles are ${player.playerBattles[0]}`);
+     *  })
+     *  .catch(error => console.log(error.message));
+     */
+    async getPlayerBattles(tag) {
+        const verifiedTag = this.verifyTag(tag);
+        const res = await this._get(`player/${verifiedTag}/battles/`);
+        return new Player({ playerBattles: res });
```
Comment (bsantosh909) [inline src/client.js:143]: You can simply name it `battles` as told below.

File: src/client.js [hunk 3]
```diff
@@ -122,6 +122,25 @@ class Client {
...
+    /**
+     * Gets the player battle history data from the API with the provided tag.
+     * @since 3.0.1
+     * @param {string} tag The player tag to get the data for.
+     * @returns {Promise<Player>} The arranged player data.
+     * @example
+     * API.getPlayerBattles('CVLQ2GV8', {
+     * })
+     *  .then(player => {
+     *    console.log(`The Player's past 25 battles are ${player.playerBattles[0]}`);
```
Comment (bsantosh909) [inline src/client.js:136]: Not so necessary but since you have taken the 0th element, your sentence gives false message 😛

File: src/client.js [hunk 4]
```diff
@@ -184,6 +203,47 @@ class Client {
...
+	 *  .then(clan => {
+	 *    console.log(clan.warLog[0]);
+	 *  })
+	 *  .catch(error => console.log(error.message));
+     */
+    async getClanWarStats(tag, options = {}) {
+        const verifiedTag = this.verifyTag(tag);
+
+        // Checking if the keys for options are correct or not.
+        console.log(options.keys);
```
Comment (bsantosh909) [inline src/client.js]: I see debugging mode.

File: src/structures/clan.js [hunk 5]
```diff
@@ -99,30 +86,46 @@ class Clan {
...
 			 */
             this.members = data.members.map(mem => new ClanMember(mem));
         }
+
+        if (data.state) {
+            /**
+			 * This is the clan's current war status.
+			 * @type {Object}
+			 */
+            this.state = data.state;
```
Comment (bsantosh909) [inline src/structures/clan.js:119]: It should be `warState` instead of state I guess.

File: src/structures/player.js [hunk 6]
```diff
@@ -199,6 +199,14 @@ class Player {
...
+            /**
+             * This player's past 25 battles
+             * @type {object<Array>}
+             */
+            this.playerBattles = data.playerBattles;
```
Comment (bsantosh909) [inline src/structures/player.js:207]: same as above

**Comment by BlackHoleFox** [comment]
Added proper array type checking to the client files, along with misc cleanup and a VSCode debugging definition.

**Pull request closed without merging**